config/software/bzip2.rb

```diff
@@ -40,7 +40,7 @@
   args = "PREFIX='#{install_dir}/embedded' VERSION='#{version}'"
 
   patch source: 'makefile_take_env_vars.patch', env: env
-  patch source: 'soname_install_dir.patch', env: env if mac_os_x_mavericks?
+  patch source: 'soname_install_dir.patch', env: env if mac_os_x?
```

Comment:
That's correct, the original patch was very conservative, see https://github.com/chef/omnibus-software/blob/master/config/patches/bzip2/soname_install_dir.patch#L7 on OSX the compiler takes different options, so this patch should apply on all versions of OSX (at least on modern ones).